### PR TITLE
Event stats

### LINKED ISSUE #190: eventAggregates should save statistics
other than eventCounts and eventPayouts, we should have eventStats, which should include counts of events, segmented by (in the key):

* device type (mobile, desktop)
* browser
* operating system
* location, using CF-IPcountry header (if any)

some events will have statistics (for now only IMPRESSIONS), and the others will not

--------------------

alternatively, we can adapt `eventCounts` for this by making the key be `[publisherAddr, adUnit, deviceType, browser, OS, location].join(SEPARATOR)`; it is already segmented by `publisherAddr` and `adUnit`, so we might as well use the same concept for stats

### PR DESCRIPTION
**Pull request by samparsky**
Fixes #190 and #191

### CODE REVIEW

File: services/sentry/eventAggregator.js [hunk 1]
```diff
@@ -51,10 +52,13 @@ function makeRecorder(channelId) {
 		leading: false,
 		trailing: true
 	})
-
 	const updateChannelPrice = async ev => {
-		const price = ev.type === 'UPDATE_IMPRESSION_PRICE' ? ev.price : '0'
-		await channelsCol.updateOne({ id: channelId }, { $set: { currentPricePerImpression: price } })
+		if (ev.type === eventTypes.IMPRESSION_PRICE_PER_CASE) {
```
Comment (Ivshti) [inline services/sentry/eventAggregator.js]: later improvement: consider merging `IMPRESSION_PRICE_PER_CASE` and `UPDATE_IMPRESSION_PRICE` into one event (considering that this is still not frozen in spec)

File: services/sentry/lib/eventReducer.js [hunk 2]
```diff
@@ -56,22 +64,41 @@ function isPriceOk(price, channel) {
...
 }
+// returns the price per impression
+function getPrice(channel, ev) {
+	let price = new BN(channel.currentPricePerImpression || channel.spec.minPerImpression, 10)
+	if (channel.pricePerImpressionCase && ev.stat) {
+		// check if there is a unique price case
+		// for the event stat
+		// it can be publisher prefixed for a specific publisher
+		const result = channel.pricePerImpressionCase.find(
+			priceCase => priceCase.stat === ev.stat || priceCase.stat === `${ev.publisher}:${ev.stat}`
```
Comment (Ivshti) [inline services/sentry/lib/eventReducer.js]: this allows setting the price per-publisher, right?

Comment (samparsky) [reply services/sentry/lib/eventReducer.js]: Yes

File: services/sentry/eventAggregator.js (hunk 1 above)
Comment (samparsky) [reply services/sentry/eventAggregator.js]: Ohh okay, which event name should it be `UPDATE_IMPRESSION_PRICE `?

Comment (Ivshti) [reply services/sentry/eventAggregator.js]: Yes, but start with a PR to the documentation with the proposed interface

Comment (Ivshti) [reply services/sentry/eventAggregator.js]: see this: https://github.com/AdExNetwork/adex-validator/issues/203

**Pull request merged**